File: programmers/c++/모의고사.cpp
//모의고사
//33분
//%= 조심히 사용하기 = 있으면 그값자체가 변하는 거 조심!! 실수 노노

#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
	vector<int> answer;
	int one[] = { 1,2,3,4,5 };
	int two[] = { 2,1,2,3,2,4,2,5 };
	int three[] = { 3,3,1,1,2,2,4,4,5,5 };
	int n1 = 5;
	int n2 = 8;
	int n3 = 10;
	int score[3] = { 0,0,0 };
	for (int i = 0; i < answers.size(); i++) {
		if (answers[i] == one[i % n1])
			score[0]++;
		if (answers[i] == two[i % n2])
			score[1]++;
		if (answers[i] == three[i % n3])
			score[2]++;
	}
	int max_score = 0;
	for (int i = 0; i < 3; i++)
		max_score = max(max_score, score[i]);
	for (int i = 0; i < 3; i++) {
		if (max_score == score[i])
			answer.push_back(i + 1);
	}
	return answer;
}

File: programmers/python/sort/모의고사.py
#첫번째 풀이

def solution(answers):
    answer = []
    score = [0,0,0]
    one = [i for i in range(1,6)]
    two = [2, 1, 2, 3, 2, 4, 2, 5]
    three = [ 3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    people = [one,two,three]
    num = 0
    for ans in answers:
        for i in range(3):
            if ans == people[i][num % len(people[i])]:
                score[i] += 1
        num += 1
    maxValue = max(score)
    for i in range(3):
        if maxValue == score[i]:
            answer.append(i+1)
    return answer

#두번째 풀이
'''
enumerate으로 더 깔끔한 코드 가능.
'''
def solution(answers):
    answer = []
    score = [0,0,0]
    one = [i for i in range(1,6)]
    two = [2, 1, 2, 3, 2, 4, 2, 5]
    three = [ 3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    people = [one,two,three]
    
    for num, ans in enumerate(answers):
        for i in range(3):
            if ans == people[i][num % len(people[i])]:
                score[i] += 1
                
    maxValue = max(score)
    for i in range(3):
        if maxValue == score[i]:
            answer.append(i+1)
    return answer
